### PR TITLE
Copter: 4.3.0-beta1 release

### PR DESCRIPTION
**Pull request by rmackay9**
This is the Copter and Rover 4.3.0-beta1 release which is master at the moment but may be rebased after tomorrow's dev call.

For the release notes I went through the last 30 pages of merged PRs ([starting about here](https://github.com/ArduPilot/ardupilot/pulls?page=30&q=is%3Apr+is%3Aclosed)) but I undoubtedly missed some enhancements and bug fixes so please feel free to comment on this PR or PM me directly with any items I've missed.  Ideally include some user understandable comment on the change that I can add to the release notes.

### CODE REVIEW

File: ArduCopter/ReleaseNotes.txt [hunk 1]
```diff
@@ -1,5 +1,88 @@
...
+    n) RC failsafe made pre-arm check (previously only triggered at arming)
+    o) RC failsafe option (see FS_OPTIONS) to continue in Guided obeyed even if GCS failsafe disabled
+    p) TKOFF_RPM_MIN checks all motors spinning before takeoff
+    q) Vibration compensation disabled in manual modes
+7) Scripting enhancements
+    a) CAN2 port bindings to allow scripts to communicate on 2nd CAN bus
+    b) ESC RPM bindings to allow scripts to report engine RPM
+    c) Gimbal bingings to allow scripts to control gimbal
+    d) Pre-arm check bindings (allows scripts to check if pre-arm checks have passed)
+    e) Restarting scripts supported (may require GCS support to be added)
```
Comment (tridge) [inline ArduCopter/ReleaseNotes.txt:59]: this was in 4.2, but memory leak on restart fixed in 4.3

File: ArduCopter/ReleaseNotes.txt [hunk 2]
```diff
@@ -1,5 +1,88 @@
...
+    f) Semicolon (:) and period (.) supported (e.g both Logger:write() and Logger.write will work)
+    g) VTOL QuickTune script allows easy/safe tuning of quadcopters and multicopters
```
Comment (tridge) [inline ArduCopter/ReleaseNotes.txt:61]: this was in 4.2

File: ArduCopter/ReleaseNotes.txt [hunk 3]
```diff
@@ -1,5 +1,88 @@
...
+8) Sensor driver enhancements
+    a) Benewake H30 radar support
+    b) BMI270 IMU performance improvements
+    c) Logging pause-able with auxiliary switch.  see RCx_OPTION = 165 (Pause Stream Logging)
+    d) Proximity sensor support for up to 3 sensors
+    e) Precision Landing consumes LANDING_TARGET MAVLink message's PositionX,Y,Z fields
+    f) RichenPower generator maintenance-required messages can be suppressed using GEN_OPTIONS param
+    g) TeraRanger Neo rangefinder support
+    h) UBlox GPS support to provide ellipsoid altitude instead of AMSL (see GPS_DRV_OPTIONS)
```
Comment (tridge) [inline ArduCopter/ReleaseNotes.txt:70]: most GPS drivers gained ellipsoid support, not just ublox

File: ArduCopter/ReleaseNotes.txt [hunk 4]
```diff
@@ -1,5 +1,88 @@
...
+    b) Cygbot proximity sensor fix to support different orientations (see PRXx_ORIENT)
+    c) Loiter fix to avoid potential wobble on 2nd takeoff (was not clearing non-zero attitude target from previous landing)
+    d) Lutan EFI message flood reduced
+    e) Missions download to GCS corruption avoided by checking serial buffer has space
+    f) Payload place fix so vehicle flies to specified Lat,Lon (if provided).  Previously it could get stuck
+    g) Safety switch disabled if IOMCU is disabled (see BRD_IO_ENABLE=0)
+    h) Takeoff vertical velocity limits enforced correctly even if PILOT_TKOFF_ALT set to a significant height
+10) Developer items
+    a) Custom controller support
+    b) Parameter defaults recorded in @SYS/defaults.parm and onboard logs
```
Comment (tridge) [inline ArduCopter/ReleaseNotes.txt:82]: there is no @SYS/defaults, its built into the param ftp protocol

File: ArduCopter/ReleaseNotes.txt (hunk 2 above)
Comment (rmackay9) [reply ArduCopter/ReleaseNotes.txt:61]: removed, txs

File: ArduCopter/ReleaseNotes.txt (hunk 1 above)
Comment (rmackay9) [reply ArduCopter/ReleaseNotes.txt:59]: moved to bug fix section, txs

File: ArduCopter/ReleaseNotes.txt (hunk 3 above)
Comment (rmackay9) [reply ArduCopter/ReleaseNotes.txt:70]: reworded to cover all GPSs.

**Comment by rmackay9** [comment]
@andyp1per, @tridge, thanks for the feedback, this has been updated now I think.

File: ArduCopter/ReleaseNotes.txt (hunk 4 above)
Comment (rmackay9) [reply ArduCopter/ReleaseNotes.txt]: ok fixed, txs

**Pull request merged**